GOVERNANCE.md

```diff
@@ -0,0 +1,136 @@
+* Reading/writing of data from/to binary files like parquet
+
+# Roles 
+
+## Contributor
+
+* Definition: a user who has written/commented at least one issue, worked to label/triage issues, written a blog post, given a talk, etc. 
+* How this role is recognized: there is no central list of contributors / no formal recognition for contributors.
+
+## Project member
+
+* Definition: some one who has submitted at least one PR with code contributions, that has been merged into master. (spelling fixes in docs do not count)
+* How to obtain this role: any user/contributor can become a member by submitting a PR with code contributions, then having it reviewed and merged into master. Contributors who have written issues should be encouraged to submit their first PR to become a project member. Contributors can look at https://github.com/Rdatatable/data.table/labels/beginner-task for easy issues to work on.
+* How this role is recognized: Members are credited via role="ctb" in DESCRIPTION (so they appear in Author list on CRAN), and they are added to https://github.com/orgs/Rdatatable/teams/project-members so they can create new branches in the Rdatatable/data.table GitHub repo. They also appear on https://github.com/Rdatatable/data.table/graphs/contributors (Contributions to master, excluding merge commits).
+
+## Reviewer
+
+* Definition: a member who has volunteered to do code reviews for some features/files. 
+* How to obtain this role: after one or more significant PRs to a given file, a member should be invited to add their name as a reviewer of that file in CODEOWNERS, and after that is merged into master, then they are considered a reviewer. 
+* How this role is recognized: same credit in DESCRIPTION as a regular member, role="ctb" (so they appear in Author list on CRAN). 
+* Note: having your name in CODEOWNERS does not give any special permission, but it does mean that you will be notified whenever there is a new PR with changes to that file. 
+
+## Committer
+
+* Definition: permission to commit to, and merge PRs into, master branch. 
+* How to obtain this role: after a reviewer has a consistent history of careful reviews of others' PRs, then a current Committer should ask all other current Committers if they approve promoting the Reviewer to Committer, and it should be done if there is Consensus among active Committers. 
+* How this role is recognized: credited via role="aut" in DESCRIPTION (so they appear in Author list on CRAN), and added to https://github.com/orgs/Rdatatable/teams/maintainers which gives permission to merge PRs into master branch. 
+
+## CRAN maintainer
+
+* Definition: in charge of communication with CRAN. Responsible for submitting releases to CRAN on a regular basis, and for responding to requests from CRAN.
+* How to obtain this role: (1) merge into master a PR adding role="cre" to DESCRIPTION, and (2) submit updated package to CRAN (previous CRAN maintainer will have to confirm change by email to CRAN). 
```

Comment:
This seems a bit ill-phrased since new Comitters need to be a consensus voting but new CRAN maintainers are just decided by the current CRAN maintainer?

Reply:
what change would you suggest?

Reply:
```suggestion
* How to obtain this role: (1) create a PR into master adding role="cre" to DESCRIPTION. This PR has to approved by consensus of committers with a voting period of at least two weeks. (2) submit updated package to CRAN (previous CRAN maintainer will have to confirm change by email to CRAN).
```

Maybe something like this? I would definitely like a lower bound on the voting period (maybe even 4 or more weeks) for major team changes like committer and CRAN maintainer since non-response counts as approval.

Reply:
two weeks seems reasonable to me. I would change to the following though, to avoid "voting period" since voting process is not defined (but consensus is).

(1) create a PR into master adding role="cre" to DESCRIPTION. This PR has to approved by Consensus of committers with review period of at least two weeks.

Reply:
I think it also make sense for Matt to be included in deciding future CRAN maintainer changes.

Reply:
I don't think we should specifically mention Matt, but Matt would definitely be included, because he is a committer. Is that sufficient?